src/LinkedItems.php

```diff
@@ -8,97 +8,122 @@
         })->toArray();
     }
 
-    public static function linkableUrls($url, $translation): array
-    {
-        return [$url => $translation];
-    }
-
     /**
-     * Builds an array of linkable items and routes to feed a list on the back office
+     * Returns an array of well-formed linkable route.
+     * Check out the config file or the readme file to know more about linkable routes.
      *
+     * @overridable
+     * @param $routeName
+     * @param $translation
      * @return array
      */
-    public static function links(): array
+    public static function linkableRoute(string $routeName, string $translation): array
     {
-        $links = [];
-        $linkableObjects = config('backpack.laravel-backpack-menu.linkableObjects', []);
-        foreach ($linkableObjects as $class => $translation) {
-            $items = $class::linkableItems(trans($translation));
-            $links = array_merge($links, $items);
-        }
-
-        $linkableUrls = config('backpack.laravel-backpack-menu.linkableUrls', []);
-        foreach ($linkableUrls as $url => $translation) {
-            $items = $class::linkableUrls($url, trans($translation));
-            $links = array_merge($links, $items);
-        }
-
-        asort($links);
-
-        return $links;
+        return [$routeName => $translation];
     }
 
-    public function linkableUrl()
+    /**
+     * Returns the url of an linkable item.
+     *
+     * @overridable
+     * @return \Illuminate\Contracts\Routing\UrlGenerator|string
+     */
+    public function linkableUrl(): string
     {
-        return url($this->slug);
+        return isset($this->slug) ? url($this->slug) : '';
     }
 
     /**
-     * Returns the label for the item. Title by default.
-     * Optionally overridable within items having no title (for instance name)
+     * Returns the label of the linkable item.
+     * Title by default.
+     *
+     * @overridable
      * @return string
      */
-    public function linkableTitle()
+    public function linkableTitle(): string
     {
-        return $this->title;
+        return isset($this->title) ? $this->title : '';
     }
 
     /**
-     * Returns the id for the item.
-     * Optionally overridable within items having a primary other than:
-     *  the default "id" primary key.
-     *  the custom primary key defined within the model.
+     * Returns the id of the linkable item.
      *
+     * @overridable
      * @return string
      */
-    public function linkableId()
+    public function linkableId(): string
     {
         $primaryKey = $this->getKeyName();
 
         return $this->{$primaryKey};
     }
 
     /**
-     * Returns an array:
+     * Returns a sorted collection of linkable items and routes.
+     * This collection is used in the back office (backpack) to feed a select list.
+     * This select list is intended for adding new menu items.
+     *
+     * @return array
+     */
+    final public static function links(): array
```

Comment:
Why do you add this `final` keyword? I think we should allow the developper to override this method in his Model.

Reply:
I am not that sure! This method contains two static calls to methods which are both overridables. 
I think this gives the developer enough freedom.

Reply:
Personally, I hate these `final`. Let me override what i want! ;-)

Reply:
Agree with @jguyomard , `final` should be deleted.